core/services/request/middleware/BotDetector.php

```diff
@@ -27,12 +26,17 @@ public function handleRequest(RequestInterface $request, ResponseInterface $resp
     {
         $this->request = $request;
         $this->response = $response;
-        /** @var CrawlerDetect $CrawlerDetect */
-        $CrawlerDetect = $this->loader->getShared('EventEspressoVendor\CrawlerDetect\CrawlerDetect');
-        if ($CrawlerDetect instanceof CrawlerDetect) {
-            // Check and record the user agent of the current 'visitor'
-            $this->request->setIsBot($CrawlerDetect->isCrawler());
-            $this->request->setUserAgent($CrawlerDetect->userAgent());
+        if (class_exists('EventEspressoVendor\CrawlerDetect\CrawlerDetect')) {
+            /** @var \EventEspressoVendor\CrawlerDetect\CrawlerDetect $CrawlerDetect */
+            $CrawlerDetect = $this->loader->getShared('EventEspressoVendor\CrawlerDetect\CrawlerDetect');
+            if ($CrawlerDetect instanceof \EventEspressoVendor\CrawlerDetect\CrawlerDetect) {
+                // Check and record the user agent of the current 'visitor'
+                $this->request->setIsBot($CrawlerDetect->isCrawler());
+                $this->request->setUserAgent($CrawlerDetect->userAgent());
+            }
+        } else {
+            $this->request->setIsBot(false);
+            $this->request->setUserAgent();
         }
```

Comment:
I'm not certain if we need this else statement. Since we don't have an else statement for the if statement on line 32 then I think we can remove this too. Implemented just to discuss it.

Reply:
it doesn't hurt... 
it at least sets those values to the proper data types, because they are potentially `null` prior to this point